### PR TITLE
[FLINK-28347][Test infrastructure] Update testcontainers dependency to v1.17.6

### PR DESCRIPTION
**Pull request by MartijnVisser**
## What is the purpose of the change

* Update Testcontainers to latest version

## Brief change log

* Updated version in POM

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes 
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Kernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by MartijnVisser** [comment]
@snuyanzin Do you want to make a review?

**Comment by snuyanzin** [comment]
Yes, sure.
LGTM as soon as tests are green

**Comment by MartijnVisser** [comment]
@flinkbot run azure

**Comment by MartijnVisser** [comment]
I'm honestly still baffled why this patch is failing on our CI while passing locally. @xcomp @snuyanzin if you have any thoughts...

**Comment by snuyanzin** [comment]
i will try to have a look including the changeset of testcontainers

**Comment by snuyanzin** [comment]
not 100% sure it seems like a kind of leak, same test is passing if it is running alone on ci
need to have a look deeper

**Comment by snuyanzin** [comment]
@MartijnVisser  seems I found the reason
It looks the issue was in inconsistent jna dependency provided by flink-end-to-end-tests.
At the same time testcontainers 1.17.6. depends on docker-java-transport-zerodep 3.2.13 which  depends on jna 5.8.0

```
[INFO] +- org.testcontainers:junit-jupiter:jar:1.17.6:test
[INFO] |  \- org.testcontainers:testcontainers:jar:1.17.6:test
[INFO] |     +- org.apache.commons:commons-compress:jar:1.21:test
[INFO] |     +- org.rnorth.duct-tape:duct-tape:jar:1.0.8:test
[INFO] |     |  \- org.jetbrains:annotations:jar:17.0.0:test
[INFO] |     +- com.github.docker-java:docker-java-api:jar:3.2.13:test
[INFO] |     |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.13.4:test
[INFO] |     \- com.github.docker-java:docker-java-transport-zerodep:jar:3.2.13:test
[INFO] |        +- com.github.docker-java:docker-java-transport:jar:3.2.13:test
[INFO] |        \- net.java.dev.jna:jna:jar:5.8.0:test

```

by the way since 5.7.0 contains fixes for M1 it could potentially improve situation with tests on M1 (I can not check it)

File: pom.xml [hunk 1]
```diff
@@ -157,7 +157,8 @@ under the License.
 		<beam.version>2.43.0</beam.version>
 		<protoc.version>3.21.7</protoc.version>
 		<okhttp.version>3.14.9</okhttp.version>
-		<testcontainers.version>1.17.2</testcontainers.version>
+		<!-- keep jna version in sync -->
```
Comment (XComp) [inline pom.xml:160]:
```suggestion
		<!-- keep net.java.dev.jna:jna version in sync -->
```
nit: shall we be a bit more specific here? I'm just afraid that the term "jna" is too generic to find the right location in a simple grep

**Review by XComp — APPROVED** [review]
Great catch, indeed. The change looks good to me except for a minor comment on the comment 8)

File: pom.xml (hunk 1 above)
Comment (snuyanzin) [reply pom.xml]: I think it makes sense, thanks for the suggestion!

**Pull request merged**